include/energy.h

```diff
@@ -32,7 +32,7 @@ namespace Energy {
      *
      * TODO(#22): generate a proper doxygen comment
      */
-    void ComputeEnergy(cv::InputOutputArray img);
+    void ComputeEnergy(cv::InputArray in_img, cv::OutputArray out_img);
```

Comment:
Changed this function from only operating in-place so I don't have to limit myself in the future.